web/client/components/map/openlayers/Map.jsx

```diff
@@ -228,14 +261,36 @@ export default class OpenlayersMap extends React.Component {
         if (newProps.mousePointer !== this.props.mousePointer) {
             this.setMousePointer(newProps.mousePointer);
         }
-
         if (newProps.zoomControl !== this.props.zoomControl) {
             if (newProps.zoomControl) {
                 this.map.addControl(new Zoom());
             } else {
                 this.map.removeControl(this.map.getControls().getArray().filter((ctl) => ctl instanceof Zoom)[0]);
             }
         }
+        /*
+         * Manage interactions programmatically.
+         * map interactions may change, i.e. becoming enabled or disabled
+         */
+        if (this.map && (this.props.mapOptions && this.props.mapOptions.interactions) !== (newProps.mapOptions && newProps.mapOptions.interactions)) {
+            const newInteractions = newProps.mapOptions.interactions || {};
+            Object.keys(newInteractions).forEach(newInteraction => {
+                this.map.getInteractions().getArray().forEach((mapInteraction) => {
+                    if (interactionsDefaults[newInteraction] && mapInteraction instanceof interactionsDefaults[newInteraction].Instance) {
+                        this[`${newInteraction}Interaction`] = mapInteraction;
```

Comment:
Use a local variable. 
Using this could cause problems if some interactions are removed, they may be still in "this" object
This is a simple "find interaction". You can event use a find from lodash on the array

Reply:
the problem is that when [this](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/map/openlayers/Map.jsx#L109) line runs it creates a bunch of interactions by default. 

we should avoid to duplicate them. i'll improve this with a find function and then adding if it not exists.

if i find the interaction then i'll have just to activate/ deactivate it